src/ngraph/descriptor/input.cpp

```diff
@@ -33,9 +33,21 @@ descriptor::Input::Input(Node* node, size_t index, Output& output)
     output.add_input(this);
 }
 
+descriptor::Input::Input(Node* node, size_t index)
+    : m_node(node)
+    , m_index(index)
+    , m_output(nullptr)
+    , m_is_relevant_to_shape(false)
```

Comment:
Maybe I shoulda called this something like `m_is_relevant_to_output_shapes`. Anyway...